### DISCUSSION TITLE
Is django-admin-sortable2 compatible with non-integer primary keys?

### DISCUSSION THREAD

**Discussion by knifecake** [jrief/django-admin-sortable2#314] [Q&A] [answered]
I'm testing it out on a project with string primary keys and I see that the async requests made by django-admin-sortable2 include the primary key casted to an int in the body. This breaks sorting because the correct primary key is not being sent and therefore the server replies with a 400 Bad Request error.

**Comment by jrief** [comment]
Please read section on how to report bug:
https://django-admin-sortable2.readthedocs.io/en/latest/contributing.html#reporting-bugs

**Reply by knifecake** [reply]
Thanks for your fast reply! I was just wondering if the project supports string primary keys.

I've verified it does not work with string primary keys using the latest code from master per your instructions, but I'm not sure if it's a bug since it could just be that string primary keys are not supported.

**Comment by jrief** [comment, marked as answer]
> I was just wondering if the project supports string primary keys.

I read somewhere (don't remember where) that string primary keys are a bad idea anyway, so I never tested. Unless you can give me a good reason, why `unique=True` on that field (plus an inter-based PK) doesn't work for you, I'll probably not going to change the code for adminsortable2.

**Reply by knifecake** [reply]
Alright, thanks for being upfront about this!
